RMSD/rmsd.cpp

```diff
@@ -0,0 +1,174 @@
+#include <stdio.h>
+#include <mpi.h>
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include <math.h>
+#include <qcprot.h>
+#include <sys/time.h>
+
+using namespace std;
+
+double rmsd(double** xref){
+
+    double **xmobile;
+    double temp;
+    double *pointer = new double[9];
+
+    xmobile = new double*[3];
+    
+    for (int i=0;i<3;i++){
+        xmobile[i] = new double[146];
+        for (int j=0;j<146;j++){
+            temp = rand();
+            xmobile[i][j] = (temp/RAND_MAX)*15;
+        }
+    }
+
+    temp = CalcRMSDRotationalMatrix(xref,xmobile,146,pointer,NULL);
+
+    return temp;
+};
+
+double *block_rmsd(double **xref0, int start, int stop,int step,int rank){
+
+    int bsize;
+    double *results;
+
+    double *t_comp;
+
+    bsize = stop - start;
+
+    results = new double[stop-start];
+    for (int i=0;i<stop-start;i+=step){
+        results[i] = rmsd(xref0);
+    }
+
+    return results;
+
+};
+
+
+int main (){
+    int rank;
+    int ierr;
+    int size;
+//
+//  Initialize MPI.
+//
+    ierr = MPI_Init ( NULL, NULL );
+//
+//  Get the number of processes.
+//
+    ierr = MPI_Comm_size ( MPI_COMM_WORLD, &size );
+//
+//  Get the individual process ID.
+//
+    ierr = MPI_Comm_rank ( MPI_COMM_WORLD, &rank );
+
+
+
+    int nframes=1024; //total number of frames
+    int bsize; // number of frames of this process
+    double *results; // pointer to the RMSD results array from all processes
+    double **xref0; //Reference Frame pointer. It is a 3 by 3341
+    int start,stop; //first and last frame that will be used in this process
+    double *result;  //pointer to RMSD result array per process.
+    long int tstart,tstop; // Variable that will hold timestamps for timing values.
+    struct timeval tp; //Time struct.
+
+
+
+    int *duration = new int[3]; //keeps track of durations for each process
+    int *durations = new int[size*3]; // Pointer that will gather all the information
+
+    // Get timestamp and convert it to ms
+    gettimeofday(&tp, NULL);
+    tstart = tp.tv_sec * 1000 + tp.tv_usec / 1000;
```

Comment:
I will suggest creating a function to get the current time and then using it. Something like:
```
float current_time(){
    gettimeofday(&tp, NULL);
    current_time = tp.tv_sec * 1000 + tp.tv_usec / 1000;
    return current_time;
}
```

Reply:
Okay! I think you are right!

Reply:
Please don't judge my c skills. That was not the point of this comment! :P